scalafix-cli/src/main/scala/scalafix/cli/Cli.scala

```diff
@@ -105,7 +106,12 @@ object Cli {
   }
 
   def zshNames: String =
-    ScalafixRules.allNames.map(x => s""""$x"""").mkString(" \\\n  ")
+    ScalafixRules.allNamesDescriptions
+      .map {
+        case (name, description) =>
+          s""""$name[${StringEscapeUtils.escapeXSI(description)}]""""
```

Comment:
It is necessary to quote characters like `|` ` &`  `;`  ... (see this [doc](http://pubs.opengroup.org/onlinepubs/7908799/xcu/chap2.html)).

I think it is reasonable to use [org.apache.commons.text.StringEscapeUtils.escapeXSI](https://github.com/apache/commons-text/blob/ddb47903dcab034e7dd33eadfe2658eb3aad0739/src/main/java/org/apache/commons/text/StringEscapeUtils.java#L842-L861).

Reply:
Sounds good 👍 If we want to slim down the jars we can always inline the implentation, doesn't look too bad https://github.com/apache/commons-text/blob/ddb47903dcab034e7dd33eadfe2658eb3aad0739/src/main/java/org/apache/commons/text/StringEscapeUtils.java#L234-L261

Reply:
Should we also escape cli flag descriptions? I've noticed that `scalafix --<TAB>` can act a bit weirdly sometimes.